packages/shared/lib/services/connection.service.ts

```diff
@@ -360,8 +368,6 @@ class ConnectionService {
             }
         }
 
-        await this.updateLastFetched(connection?.id as number);
```

Comment:
When `getConnection` is called it actually isn't fetching credentials. The correct place is `getConnectionDetails`